nnvm/src/top/tensor/transform.cc

```diff
@@ -128,15 +128,23 @@ inline bool ConcatenateCorrectLayout(const NodeAttrs& attrs,
                                      std::vector<Layout> *ilayouts,
                                      const std::vector<Layout> *last_ilayouts,
                                      std::vector<Layout> *olayouts) {
+  const ConcatenateParam& param = nnvm::get<ConcatenateParam>(attrs.parsed);
```

Comment:
didn't think of it carefully, it will produce incorrect results when, e.g., axis=1, last_layout=NCHW, ilayout=NHWC
actually need to at least check `ilayout[axis] == last_layout[axis]`. but not quite sure whether NCHW -> N16cHWC and axis=1 can also be correct.

Reply:
OK. I think we need to check: ilayout param.axis has the same letter with last_layout, and it should be a plain capital letter(C but not 4c or 16c).  C means the actual number can vary, which is the definition for concatenate. For NCHW -> N16cHWC and axis=1, it is invalid since this transformation can't guarantee the actual number of last C is the same across all inputs.